### PR TITLE
Added documentation for printing associated queries, and sample documents in notification message.

### PR DESCRIPTION
**Pull request by AWSHurneyt**
### Description
1. Changed the examples to access items in `ctx` arrays from using brackets (e.g., `itemList[0]`) as mustache template uses dot notation (e.g., `itemList.0`).
2. Added documentation for `associated_queries`, and `sample_documents` variables added to the `ctx` object.

Implementation PR for reference - https://github.com/opensearch-project/alerting/pull/1450

### Issues Resolved
https://github.com/opensearch-project/alerting/issues/1300
https://github.com/opensearch-project/alerting/issues/1396
https://github.com/opensearch-project/alerting/issues/1401

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by AWSHurneyt** [comment]
Draft discussion forum post - https://forum.opensearch.org/t/alerting-sample-documents-for-ctx-dedupedalerts-and-ctx-completedalerts/18695

**Comment by hdhalter** [comment]
Thanks so much for adding this content, @AWSHurneyt! Is it ready for a doc review?

**Comment by AWSHurneyt** [comment]
> Thanks so much for adding this content, @AWSHurneyt! Is it ready for a doc review?

@hdhalter Yes, this can be reviewed. I actually have a few other updates I was considering adding to this PR as well. We're still waiting for final approval though, so I'll just create a separate PR for them. Those updates would have be unrelated to the feature described in these doc changes; so it's probably for the best to keep them separate.

File: _observing-your-data/alerting/triggers.md
Comment (AWSHurneyt) [inline _observing-your-data/alerting/triggers.md:188]: **Question for documentation reviewers:**
The following on line 146, calls out that the variable names for the "alerts" list is `newAlerts` for per bucket monitors, and `alerts` for per document monitors.
> Each new alert that's generated during a monitor execution will be added to a list (i.e., `newAlerts` for per bucket monitors, and `alerts` for per document monitors) within the `ctx` variable.

This example is for a per document monitor as the "alerts" list is access via `{{#ctx.alerts}}` instead of `{{#ctx.newAlerts}}`. Does that seem clear enough?

File: _observing-your-data/alerting/triggers.md
Comment (Naarcha-AWS) [inline _observing-your-data/alerting/triggers.md:139]:
```suggestion
`bucket_keys` | String | A comma-separated list of the monitor's bucket key values. Available only for `ctx.dedupedAlerts`, `ctx.newAlerts`, and `ctx.completedAlerts`. Accessed through `ctx.dedupedAlerts.0.bucket_keys`.
```

File: _observing-your-data/alerting/triggers.md
Comment (Naarcha-AWS) [inline _observing-your-data/alerting/triggers.md:146]:
```suggestion
Per bucket and per document monitors support printing sample documents in notification messages. Per document monitors support printing the list of queries that triggered the creation of the finding associated with the alert. When the monitor runs, the monitor adds each new alert that's generated into the within the `ctx` variables, for example, `newAlerts` for per bucket monitors and `alerts` for per document monitors.  Each alert has its own list of `sample_documents` and each per document monitor alert has its own list of `associated_queries`. The message template can be formatted to iterate through the list of alerts, the list of `associated_queries`, and the `sample_documents` for each alert.
```

File: _observing-your-data/alerting/triggers.md
Comment (Naarcha-AWS) [inline _observing-your-data/alerting/triggers.md:154]:
```suggestion
`_index` | String | The index containing the sample document.
```

File: _observing-your-data/alerting/triggers.md
Comment (Naarcha-AWS) [reply _observing-your-data/alerting/triggers.md:188]: Lets update line 184 and call it out.

File: _observing-your-data/alerting/triggers.md
Comment (Naarcha-AWS) [inline _observing-your-data/alerting/triggers.md:184]:
```suggestion
The `_source` object in this example is based on the `opensearch_dashboards_sample_data_ecommerce` index that's available on OpenSearch Dashboards.
{: .note}
```

File: _observing-your-data/alerting/triggers.md
Comment (Naarcha-AWS) [inline _observing-your-data/alerting/triggers.md:135]:
```suggestion
`ctx.alerts` | Array | Newly created alerts. Includes the `ctx.alerts.0.finding_ids` that triggered the alert and the `ctx.alerts.0.related_doc_ids` associated with the findings. Only available with document-level monitors.
```

File: _observing-your-data/alerting/triggers.md
Comment (Naarcha-AWS) [inline _observing-your-data/alerting/triggers.md:138]: What do we mean by "no longer ongoing"? Have these alerts been disabled automatically or manually?

```suggestion
`ctx.completedAlerts` | Array | Alerts that are no longer ongoing. Only available with bucket-level monitors.
```

File: _observing-your-data/alerting/triggers.md
Comment (Naarcha-AWS) [inline _observing-your-data/alerting/triggers.md:139]:
```suggestion
`bucket_keys` | String | A comma-separated list of the monitor's bucket key values. Available only for `ctx.dedupedAlerts`, `ctx.newAlerts`, and `ctx.completedAlerts`. Accessed through the `ctx.dedupedAlerts.0.bucket_keys` setting.
```

File: _observing-your-data/alerting/triggers.md
Comment (Naarcha-AWS) [inline _observing-your-data/alerting/triggers.md:141]:
```suggestion
`associated_queries` | Array | An array of document-level monitor queries that triggered the creation of the finding associated with the alert. Only available with document-level monitors. Accessed through the `ctx.alerts.0.associated_queries` setting.
```

File: _observing-your-data/alerting/triggers.md
Comment (Naarcha-AWS) [inline _observing-your-data/alerting/triggers.md:142]:
```suggestion
`sample_documents` | Array | An array of sample documents that matched the monitor query. Only available with bucket-level, and document-level monitors. Accessed through `ctx.newAlerts.0.sample_documents` and `ctx.alerts.0.sample_documents` settings, respectively.
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

File: _observing-your-data/alerting/triggers.md
Comment (AWSHurneyt) [reply _observing-your-data/alerting/triggers.md:139]: @Naarcha-AWS Oh these aren't settings. Essentially, the `ctx` variable is a JSON payload. Mustache template uses dot notation to access the key/value pairs in the JSON. Rather than referring to this as a "setting," I think "variable" would make more sense.

File: _observing-your-data/alerting/triggers.md
Comment (AWSHurneyt) [reply _observing-your-data/alerting/triggers.md:141]: Same comment as https://github.com/opensearch-project/documentation-website/pull/6866/files#r1594326763

File: _observing-your-data/alerting/triggers.md
Comment (AWSHurneyt) [reply _observing-your-data/alerting/triggers.md:142]: Same comment as https://github.com/opensearch-project/documentation-website/pull/6866/files#r1594326763

File: _observing-your-data/alerting/triggers.md
Comment (AWSHurneyt) [reply _observing-your-data/alerting/triggers.md:146]: @Naarcha-AWS Could you clarify this line? I think there's a typo.
> *that's generated into the within the `ctx` variables*

File: _observing-your-data/alerting/triggers.md
Comment (AWSHurneyt) [reply _observing-your-data/alerting/triggers.md:190]: Revised.

File: _observing-your-data/alerting/triggers.md
Comment (Naarcha-AWS) [reply _observing-your-data/alerting/triggers.md:146]: Should be "into the `ctx` variable:

File: _observing-your-data/alerting/triggers.md
Comment (Naarcha-AWS) [reply _observing-your-data/alerting/triggers.md:139]: Updated.

**Review by Naarcha-AWS — APPROVED** [review]

File: _observing-your-data/alerting/triggers.md
Comment (natebower) [inline _observing-your-data/alerting/triggers.md:129]:
```suggestion
`ctx.results` | Array | An array with one element (`ctx.results.0`). Contains the query results. This variable is empty if the trigger is unable to retrieve results. See `ctx.error`.
```

File: _observing-your-data/alerting/triggers.md
Comment (natebower) [inline _observing-your-data/alerting/triggers.md:136]: "Alerts that have been triggered" => "Triggered alerts"? "creating endless numbers of the same alert" => "perpetually creating the same alert"?

File: _observing-your-data/alerting/triggers.md
Comment (natebower) [inline _observing-your-data/alerting/triggers.md:142]:
```suggestion
`sample_documents` | Array | An array of sample documents that matched the monitor query. Only available with bucket-level, and document-level monitors. Accessed through the `ctx.newAlerts.0.sample_documents` and `ctx.alerts.0.sample_documents` variables, respectively.
```

Comment (natebower) [inline _observing-your-data/alerting/triggers.md:142]: Second sentence: "Available with both bucket- and document-level monitors"?

File: _observing-your-data/alerting/triggers.md
Comment (natebower) [inline _observing-your-data/alerting/triggers.md:144]:
```suggestion
#### The `associated_queries` and `sample_documents` variables
```

File: _observing-your-data/alerting/triggers.md
Comment (natebower) [inline _observing-your-data/alerting/triggers.md:146]:
```suggestion
Per bucket and per document monitors support printing sample documents in notification messages. Per document monitors support printing the list of queries that triggered the creation of the finding associated with the alert. When the monitor runs, it adds each new alert to the `ctx` variables, for example, `newAlerts` for per bucket monitors and `alerts` for per document monitors. Each alert has its own list of `sample_documents`, and each per document monitor alert has its own list of `associated_queries`. The message template can be formatted to iterate through the list of alerts, the list of `associated_queries`, and the `sample_documents` for each alert.
```

Comment (natebower) [inline _observing-your-data/alerting/triggers.md:146]: Confirm that my changes to the third sentence retain the intended meaning.

File: _observing-your-data/alerting/triggers.md
Comment (natebower) [inline _observing-your-data/alerting/triggers.md:148]:
```suggestion
An Alerting monitor uses the permissions of the user that created it. Be mindful of the Notifications plugin channel to which alert messages are sent and the content of the message mustache template. To learn more about security in the Alerting plugin, see [Alerting security](https://opensearch.org/docs/latest/observing-your-data/alerting/security/).
```

Comment (natebower) [inline _observing-your-data/alerting/triggers.md:148]: "An Alerting plugin monitor"?

File: _observing-your-data/alerting/triggers.md
Comment (natebower) [inline _observing-your-data/alerting/triggers.md:151]:
```suggestion
#### Sample document variables
```

File: _observing-your-data/alerting/triggers.md
Comment (natebower) [inline _observing-your-data/alerting/triggers.md:156]:
```suggestion
`_id` | String | The sample document ID.
```

Comment (natebower) [inline _observing-your-data/alerting/triggers.md:156]: Confirm technical meaning is retained.

File: _observing-your-data/alerting/triggers.md
Comment (natebower) [inline _observing-your-data/alerting/triggers.md:157]:
```suggestion
`_score` | Float | A positive 32-bit floating-point number illustrating the relevance of the returned document.
```

File: _observing-your-data/alerting/triggers.md
Comment (natebower) [inline _observing-your-data/alerting/triggers.md:175]:
```suggestion
#### Associated query variables
```

**Review by natebower — COMMENTED** [review]
@AWSHurneyt @Naarcha-AWS Please see my comments and changes and let me know if you have any questions. Thanks!

File: _observing-your-data/alerting/triggers.md
Comment (Naarcha-AWS) [inline _observing-your-data/alerting/triggers.md:136]:
```suggestion
`ctx.dedupedAlerts` | Array | Triggered alerts. OpenSearch keeps the existing alert to prevent the plugin from perpetually creating the same alert. Only available with bucket-level monitors.
```

File: _observing-your-data/alerting/triggers.md
Comment (Naarcha-AWS) [reply _observing-your-data/alerting/triggers.md:138]: @AWSHurneyt: Let us know if you have a take on this. I'm inclined to make it inline with the name of the variable, "Alerts that are completed or expired."

File: _observing-your-data/alerting/triggers.md
Comment (Naarcha-AWS) [reply _observing-your-data/alerting/triggers.md:148]: How about "an alerting monitor?"

File: _observing-your-data/alerting/triggers.md
Comment (natebower) [reply _observing-your-data/alerting/triggers.md:138]: "Completed or expired alerts" 😄

File: _observing-your-data/alerting/triggers.md
Comment (natebower) [reply _observing-your-data/alerting/triggers.md]: That's fine. It was the capitalization of "Alerting" that was throwing me off.

File: _observing-your-data/alerting/triggers.md
Comment (Naarcha-AWS) [inline _observing-your-data/alerting/triggers.md:148]:
```suggestion
An alerting monitor uses the permissions of the user that created it. Be mindful of the Notifications plugin channel to which alert messages are sent and the content of the message mustache template. To learn more about security in the Alerting plugin, see [Alerting security](https://opensearch.org/docs/latest/observing-your-data/alerting/security/).
```

**Pull request merged**
